packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { clamp, useControllableState, useEventCallback } from '@fluentui/react-utilities';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import { colorSliderCSSVars } from './useColorSliderStyles.styles';
+import type { ColorSliderState, ColorSliderProps } from './ColorSlider.types';
+
+const { sliderProgressVar, sliderDirectionVar, thumbColorVar } = colorSliderCSSVars;
+
+const MAX_COLOR_HUE = 360;
+
+const getPercent = (value: number, min: number, max: number) => {
+  return max === min ? 0 : ((value - min) / (max - min)) * 100;
+};
+
+export const useColorSliderState_unstable = (state: ColorSliderState, props: ColorSliderProps) => {
+  'use no memo';
+
+  const { dir } = useFluent();
+  const { min = 0, max = MAX_COLOR_HUE, onChange, value } = props;
+
+  const [currentValue, setCurrentValue] = useControllableState({
+    state: value,
+    initialState: 0,
+  });
+  const clampedValue = clamp(currentValue, min, max);
+  const valuePercent = getPercent(clampedValue, min, max);
+
+  const inputOnChange = state.input.onChange;
+
+  const _onChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(ev => {
+    const newValue = Number(ev.target.value);
+    setCurrentValue(clamp(newValue, min, max));
+
+    if (inputOnChange && inputOnChange !== (onChange as unknown as React.ChangeEventHandler<HTMLInputElement>)) {
+      inputOnChange(ev);
+    } else if (onChange) {
+      onChange(ev, { type: 'change', event: ev, value: newValue });
+    }
+  });
+
+  const rootVariables = {
+    [sliderDirectionVar]: state.vertical ? '180deg' : dir === 'ltr' ? '-90deg' : '90deg',
+    [sliderProgressVar]: `${valuePercent}%`,
+    [thumbColorVar]: `hsl(${clampedValue}, 100%, 50%)`,
```

Comment:
Currently this color is based on hue, but when more channels added, it'll be changed.